File: lista03W/zad1/Kolo.java
public class Kolo extends Figura {
  private double promien;
  
  public Kolo(double promien) {
    this.promien = promien;
  }
  
  public double getPole() {
    return Math.PI * promien * promien;
  }
  
  public double getObwod() {
    return 2 * Math.PI * promien;
  }
  
  public String getNazwa() {
    return "Kolo";
  }
}

File: lista03W/zad2/kolo.cpp
#include "kolo.hpp"

Kolo::Kolo(double r) {
  if(r <= 0)
    throw std::invalid_argument("Promien musi byc dodatni");

  promien = r;
}

double Kolo::getPole() const { 
  return M_PI * promien * promien;
}
double Kolo::getObwod() const {
  return 2 * M_PI * promien;
}
std::string Kolo::getNazwa() const {
  return "Kolo";
}